src/dune/opam_create.ml

```diff
@@ -114,6 +114,30 @@ let insert_dune_dep depends dune_version =
       dune_dep :: List.rev acc
     | (dep : Package.Dependency.t) :: rest ->
       if Package.Name.equal dep.name dune_name then
+        let get_version s =
+          match String.split_on_char ~sep:'.' s with
+          | major :: minor :: _ -> (int_of_string major, int_of_string minor)
+          | [ major ] -> (int_of_string major, 0)
+          | [] -> dune_version
+        in
+        let () =
+          match dep.constraint_ with
+          | Some (Uop (_, QVar version)) when get_version version < dune_version
+            ->
+            User_warning.emit
+              ~hints:
+                [ Pp.textf "Set dune constraint to >= %s"
```

Comment:
not that currently if `(lang dune ...)` is set to 2.6 or above, and the constraint is set, dune won't do anything clever and will just add two constraints (i.e. will become `"dune" {>= "2.6" & >= "2.6"}`). It's not wrong but visually annoying imo, so maybe we could add something for that or simply say "remove the constraint if `(dune lang ...)` is set to 2.6 or above". Idk. I don't mind too much either way.